internal/ascprails/rails_test.go

```
@@ -0,0 +1,534 @@
+	"github.com/gnanam1990/flowops/pkg/escrowcall"
+	"github.com/gnanam1990/flowops/pkg/purchasespec"
+	"github.com/gnanam1990/flowops/pkg/sellerquote"
+	x402types "github.com/x402-foundation/x402/go/v2/types"
+)
+
+func TestDispatchCapturesThenFinalizesWithoutSecondEgress(t *testing.T) {
+	fixture := railsFixture(t)
+	store := newMemoryStore(fixture.now)
+	chain := &fakeChain{observations: []ChainObservation{fixture.observation, {Timestamp: uint64(fixture.now.Unix() + 5), EvidenceDigest: testHash("32"), ObservedAt: fixture.now}}}
+	body := []byte(`{"result":"ok"}`)
+	transport := &fakeRestrictedTransport{responses: []*http.Response{successResponse(t, fixture, body)}}
```

golangci-lint (2.12.2)

[error] response body must be closed

(bodyclose)